tests/tools.vitruv.applications.pcmjava.tests/src/tools/vitruv/applications/pcmjava/tests/pcm2java/Pcm2JavaTransformationTest.xtend

```diff
@@ -0,0 +1,205 @@
+		changePcmView[
+			var repository = RepositoryFactory.eINSTANCE.createRepository();
+			repositoryInitalization.apply(repository)
+			registerRoot(repository, getProjectModelPath(repository.entityName, PCM_REPOSITORY_FILE_EXTENSION).uri)
+		]
+	}
+
+	protected def void createRepository(String repositoryName) {
+		createRepository[
+			entityName = repositoryName;
+		]
+	}
+
+	private def void createSystem((System)=>void systemInitialization) {
+		changePcmView[
+			val system = SystemFactory.eINSTANCE.createSystem();
+			systemInitialization.apply(system)
+			registerRoot(system, getProjectModelPath(system.entityName, PCM_SYSTEM_FILE_EXTENSION).uri)
+		]
+	}
+
+	protected def void createSystem(String systemyName) {
+		createSystem [
+			entityName = systemyName;
+		]
+	}
+
+	// === assertions ===
+	protected def void assertExistenceOfCompilationUnitsDeeplyEqualTo(View view,
+		Collection<CompilationUnit> expectedCompilationUnits) {
+		val equalityHelper = new EcoreUtil.EqualityHelper()
+		var allActualCompilationUnits = getJavaCompilationUnits(view)
+
+		assertEquals(expectedCompilationUnits.size, allActualCompilationUnits.size)
+		assertFalse(allActualCompilationUnits.map[name].hasDuplicate)
+		assertFalse(expectedCompilationUnits.map[name].hasDuplicate)
+
+		ignoreComparisonOfJavaUtilClassifieres(equalityHelper, allActualCompilationUnits, expectedCompilationUnits)
+		expectedCompilationUnits.forEach [
+			val actualUnit = claimJavaCompilationUnit(view, it.name)
+			adaptClassMemberOrderOfExpectedToActualOrder(it, actualUnit)
+			assertTrue(equalityHelper.equals(actualUnit, it), modelComparisonErrorMessage(it, actualUnit))
+		]
+	}
+
+	protected def void assertExistenceOfPackagesDeeplyEqualTo(View view, Collection<Package> expectedPackages) {
+		val equalityHelper = new EcoreUtil.EqualityHelper()
+		val allActualPackages = getJavaPackages(view)
+
+		assertEquals(expectedPackages.size, allActualPackages.size)
+		assertFalse(allActualPackages.map[name].hasDuplicate)
+		assertFalse(expectedPackages.map[name].hasDuplicate)
+
+		expectedPackages.forEach [
+			var actualPackage = claimJavaPackage(view, it.name)
+			assertTrue(equalityHelper.equals(actualPackage, it), modelComparisonErrorMessage(it, actualPackage))
+		]
+	}
+
+	private def String modelComparisonErrorMessage(EObject expected, EObject actual) {
+		var printTarget = new DefaultPrintTarget()
+		var printIdProvider = new DefaultPrintIdProvider()
+		var modelPrinter = new DefaultModelPrinter()
+
+		printTarget.print("Expected actual model to equal expected model.")
+		printTarget.newLine
+		printTarget.print("Actual:")
+		printTarget.newLine
+		modelPrinter.printObject(printTarget, printIdProvider, actual)
+
+		printTarget.newLine
+		printTarget.print("Expected:")
+		printTarget.newLine
+		modelPrinter.printObject(printTarget, printIdProvider, expected)
+
+		return printTarget.toString
+	}
+
+	private def void ignoreComparisonOfJavaUtilClassifieres(EqualityHelper equalityHelper,
```

Comment:
Could you add a small comment why it is needed and what you do here? I.e. the behavior of `put` on the equality helper is not documented so describe what the intention behind it is.